### PR TITLE
Validate Image Upgrade with backup and restore

### PR DESCRIPTION
**Pull request by Madanagopal19**
This PR has, 

1. Testcase UpgradeDataServiceImageAndScaleUpDsWithBackUpRestore

**Which issue(s) this PR fixes** (optional)
Closes # DS-9483

Jenkins logs - https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds2.0-byoc-branch-build-test/735/

### CODE REVIEW

File: drivers/unifiedPlatform/stworkflows/pds/workflowDataservice.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ func (wfDataService *WorkflowDataService) DeployDataService(ds dslibs.PDSDataSer
 		}
 	}
 
+	log.Debugf("waiting for a minute before taking backup...")
```
Comment (atrivedi-px) [inline drivers/unifiedPlatform/stworkflows/pds/workflowDataservice.go:74]: Please revert this change

File: tests/unifiedPlatform/pds2/pds_basic_test.go [hunk 2]
```diff
@@ -140,19 +140,32 @@ var _ = BeforeSuite(func() {
 		log.Debugf("TenantId [%s]", WorkflowTargetCluster.Project.Platform.TenantId)
 		WorkflowCc.Platform = WorkflowPlatform
 		WorkflowCc.CloudCredentials = make(map[string]platform.CloudCredentialsType)
-		cc, err := WorkflowCc.CreateCloudCredentials(NewPdsParams.BackUpAndRestore.TargetLocation)
-		log.FailOnError(err, "error occured while creating cloud credentials")
-		for _, value := range cc.CloudCredentials {
+		// cc, err := WorkflowCc.CreateCloudCredentials(NewPdsParams.BackUpAndRestore.TargetLocation)
```
Comment (atrivedi-px) [inline tests/unifiedPlatform/pds2/pds_basic_test.go:143]: Please revert this

File: tests/unifiedPlatform/pds2/pds_basic_test.go [hunk 3]
```diff
@@ -140,19 +140,32 @@ var _ = BeforeSuite(func() {
...
-		wfbkpLoc, err := WorkflowbkpLoc.CreateBackupLocation(PDSBucketName, NewPdsParams.BackUpAndRestore.TargetLocation)
-		log.FailOnError(err, "error while creating backup location")
-		log.Infof("wfBkpLoc id: [%s]", wfbkpLoc.BkpLocation.BkpLocationId)
-		log.Infof("wfBkpLoc name: [%s]", wfbkpLoc.BkpLocation.Name)
+		//wfbkpLoc, err := WorkflowbkpLoc.CreateBackupLocation(PDSBucketName, NewPdsParams.BackUpAndRestore.TargetLocation)
+		//log.FailOnError(err, "error while creating backup location")
+		//log.Infof("wfBkpLoc id: [%s]", wfbkpLoc.BkpLocation.BkpLocationId)
+		//log.Infof("wfBkpLoc name: [%s]", wfbkpLoc.BkpLocation.Name)
+		// wfbkpLoc, err := WorkflowbkpLoc.CreateBackupLocation(PDSBucketName, NewPdsParams.BackUpAndRestore.TargetLocation)
+		WorkflowbkpLoc.BkpLocation = platform.BkpLocationType{
```
Comment (atrivedi-px) [inline tests/unifiedPlatform/pds2/pds_basic_test.go:162]: Please revert this

File: drivers/unifiedPlatform/stworkflows/pds/workflowDataservice.go (hunk 1 above)
Comment (Madanagopal19) [reply drivers/unifiedPlatform/stworkflows/pds/workflowDataservice.go]: done

File: tests/unifiedPlatform/pds2/pds_basic_test.go (hunk 3 above)
Comment (Madanagopal19) [reply tests/unifiedPlatform/pds2/pds_basic_test.go]: done

File: tests/unifiedPlatform/pds2/pds_basic_test.go (hunk 2 above)
Comment (Madanagopal19) [reply tests/unifiedPlatform/pds2/pds_basic_test.go]: done

**Review by atrivedi-px — APPROVED** [review]
LGTM
